src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -323,6 +323,12 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
             {
                 var propertyName = property.Name;
                 var propertyMetadata = bindingContext.ModelMetadata.Properties[propertyName];
+                if (propertyMetadata == null)
+                {
+                    // Skip indexer properties and others ModelMetadata ignores.
+                    continue;
+                }
```

Comment:
This is a consequence of #1712. I think it's good to add the workaround here for now, I'll be fixing it this week.